tron/core/actionrun.py

```diff
@@ -327,10 +327,13 @@ def _done(self, target, exit_status=0):
             return self.machine.transition(target)
 
     def retry(self):
-        # kill if in flight
         if self.retries_remaining is None or self.retries_remaining <= 0:
             self.retries_remaining = 1
-        self.kill(final=False)
+
+        if self.is_done:
```

Comment:
Have you thought about what exit_status and exit_statuses will be in these two cases? And how that will look to the developer from tronview.

Reply:
yeah good point, this is not the cleanest form, i'll change it to not use `fail`